### ISSUE TITLE
Search on Desktop fails

### ISSUE THREAD

**Issue by dsghub** [mate-desktop/caja#1694]
#### Expected behaviour
Pressing CTRL+F on Desktop should open a search window

#### Actual behaviour
Error message pops up
Could not diplsay "x-caja-search://1/"
Caja has no installed viewer capable of displaying the folder.

#### Steps to reproduce the behaviour
1. Go to Desktop
2. Press CTRL+F

#### MATE general version
1.26.0

#### Package version
1.26.0-linuxmint1

#### Linux Distribution
Linux Mint 20.3 Una 64-bit

#### Link to bugreport of your Distribution (requirement)
https://github.com/linuxmint/mintdesktop/issues/57

**Comment by lukefromdc** [comment]
Confirmed your results, but this may be intentional: in a navigation window (which can have the desktop as a directory) the pathbar (below the main toolbar) replaces the normal path buttons with the search entry box when this is activated. The desktop however is a caja window with no toolbars (or menubars, sidepane etc) so has no place to display this. 

Implementing this on the desktop would require new additional code to either open a caja navigation window set to the desktop as its working directory, or else a dedicated search dialog

**Comment by dsghub** [comment]
It used to work flawlessly in all Linux Mint LTS versions for the last  ~10 years until 20.3. Something was broken with 20.3.

There was a tiny little window that opened up when I pressed CRT+F on Desktop. I entered a search term, and all matching icons would be highlighted. Very simple, very effective.

**Comment by lukefromdc** [comment]
Was that in Nemo by any chance? Mint's primary desktop is Cinnamon, which uses Nemo as its file manager.

**Comment by dsghub** [comment]
> Was that in Nemo by any chance? Mint's primary desktop is Cinnamon, which uses Nemo as its file manager. If this is in Nemo, the commit that added it could probably be ported to caja

No, I have been using MATE and caja for years (LTS versions). Desktop search was a feature that was always present.

**Comment by lukefromdc** [comment]
There are two different search functions in Caja. I was thinking of the magnifying glass search icon that changes the pathbar to a search bar. Type to search pops up that little box in navigation windows. I don't think I ever invoked that intentially on the desktop however. Note that I have limited experience using keyboard shortcuts

**Comment by lukefromdc** [comment]
There are two different search functions in Caja. I was thinking of the magnifying glass search icon that changes the pathbar to a search bar. Type to search pops up that little box in navigation windows.  CTRL+F in a navigation window works the same as the magnifying glass search icon, that method would not work on the desktop. I never intentionlly invoked a search on the desktop window (I do that from navigation windows) so I had no idea this ever worked. 

What was the LAST version of MATE and caja where you had this working?

**Comment by dsghub** [comment]
Probably Linux Mint 18.x. I tried t see what caja version 18.3 has, but the release notes on linuxmint.com read 404.

According to a post on forums.linuxmint.com, 18.3 had caja 1.18.5.
According to another post, 18 hat caja 1.14.1-1+sarah.

Is there an easy way to see what the last major upgrade on a system was?

**Comment by dsghub** [comment]
However, is that really the important question? It may be, but I'm not sure. Apprently, pressing CTRL+F while on Desktop is *supposed* to yield some search entity. Otherwise I would not get the error message "Could not diplsay "x-caja-search://1/"", would I?

**Comment by lukefromdc** [comment]
I am going to let other team members weigh in on this for the moment

**Comment by mycophobia** [comment]
Bumping this because this definitely shouldn't happen. Ctrl+F has this behavior in open folders in spatial mode as well. It's quite unhelpful! It should probably start "MATE Search Tool" starting from your current directory instead. (Debian 12, MATE 1.26.0, Caja 1.26.1)

**Comment by dsghub** [comment]
Thank you, @mycophobia . I have edited my original post to add that Debian 12 and Caja 1.26.1 are affected, also I have added Linux Mint 21.1. The problem persists there, because the MATE version is the same.

@lukefromdc Can we have the "confirmed" label back, please?

**Comment by lukefromdc** [comment]
Must have been removed by accidenty given that I was ever able to confirm it

**Comment by lukefromdc** [comment]
Just RECONFIRMED this on Caja from git master as of 9-6-2023, released as version 1.27.2

**Comment by mycophobia** [comment]
I've thrown together a pretty hacky workaround (I'm not a very good programmer) for this if anyone's interested:
[https://mycophobia.org/spatial_search_workaround.PATCH](url)
**Apply to caja-spatial-window.c**
It doesn't exhibit the behavior described above, but with this at least Ctrl+F on desktop and in spatial windows, along with selecting Places>Search for Files... in spatial windows will do, y'know, _something_, which in this case is pull up mate-search-tool for the current directory if done in a spatial window and for your home folder if done on the desktop.

I will also point out that Ctrl+S on the desktop as well as in spatial windows will do something similar to the expected functionality described above. You can enter a pattern and it will highlight all matching icons.

**Comment by lukefromdc** [comment]
Link to patch is incomplete, goes to `https://github.com/mate-desktop/caja/issues/url` which goes nowhere

**Comment by mycophobia** [comment]
oh thanks github. edited

**Comment by lukefromdc** [comment]
just typing on the desktop does open the tiny seach window if you first click on the desktop. Search entry will be on bottom right

**Comment by lukefromdc** [comment]
I can confirm that your patch works, and that if mate-search-tool is not installed, caja doesn't crash, rather there is no response fromm Ctrl-F. Type to seach after click on desktop works same as before.

Not sure what the others on the team want to do with this

**Comment by Ben-Fenner** [comment]
I'm also experiencing this. I can't search within directory windows either. I can provide more details if needed. Hoping this gets fixed. :+1:

**Comment by lukefromdc** [comment]
Interestingly, when running from current git master under wayland, <ctrl>-F with a navigation window focussed will turn the location bar into the searchbox, same as clicking on the magnifying glass icon. I will check to see if that is working on x11 as well.
